### PR TITLE
Add dynamic attribute bindings

### LINKED ISSUE #21: Use dynamic attributeBindings to avoid manually specifying every possible attribute.
Specifying each DOM attribute that any possible consumer will want is pretty annoying.  See [here for an example](https://github.com/dockyard/ember-one-way-controls/blob/515c01a7eb6498b47915514e235f4ce0088a6995/addon/components/one-way-input.js#L12-L37), ugh :crying_cat_face:.  This also has the negative impact of doing extra work when not needed (for supporting attributes that will never be used).

We could however, dynamically update `attributeBindings` in order to allow any attribute without us having to list them all.

That might look like this:

```js
// a blacklist for attributes that we are asking for folks to pass us in `attrs` (like `update` or `_sanitizeInput` functions for {{one-way-input}})
const attributeBlacklist = {
  update: true,
  _sanitizeInput: true
};

export default Ember.Component.extend({
  init() {
    this._super(...arguments);

    for (let key in this.attrs) {
        if (!attributeBlacklist[key]) {
          this.attributeBindings.push(key);  
        }
      }
  }
});
```

Demo Twiddle: https://ember-twiddle.com/dd0dd9b48140531ccccb

### PR DESCRIPTION
**Pull request by poteto**
Closes #21

Also closes #14 and closes #15.

### CODE REVIEW

**Comment by poteto** [comment]
cc @rwjblue

File: README.md [hunk 1]
```diff
@@ -24,16 +24,24 @@ The input can also be used as a checkbox:
...
 }
 ```
 
-This means that the `onenter` and `onescape` actions will fire if their corresponding key codes are received in the `keyUp` event.
+This means that the `onenter` and `onescape` actions will fire if their corresponding key codes are received in the `keyUp` event. 
+
+If you have the `hash` helper available ([polyfill for < Ember 2.3 here](https://github.com/cibernox/ember-hash-helper-polyfill)), you can specify your own key events in the template:
+
+```hbs
+{{one-way-input keyEvents=(hash 13="onenter" 27="onescape" 8="onbackspace")}}
```
Comment (rwjblue) [inline README.md]: Wouldn't we want to do this?

```
{{one-way-input keyEvents=(hash 13=(action "onenter") 27=(action "onescape") 8=(action "onbackspace"))}}
```

**Comment by rwjblue** [comment]
This looks great to me.  I left a small comment RE the README changes for `keyEvents` (which is actually somewhat unrelated to this PR), but feel free to disregard those comments here (we can address separately).

**Comment by poteto** [comment]
Updated as per your comment! (Let me know if the formatting of that multiline `hash` looks weird to you)

**Comment by rwjblue** [comment]
Updates to README look good, but I think I was confusing in my comment.  This syntax does not work yet, but I think it should.  We currently expect the value in `keyEvents` to be a string method name that we look up in `this.attrs`.

**Comment by cibernox** [comment]
I need to bind the style attribute of the input. Can the dynamic attr bindings be splitted from the dynamic key events and land them separately?

File: addon/components/one-way-input.js [hunk 2]
```diff
@@ -89,6 +66,12 @@ export default Component.extend({
   init() {
     this._super(...arguments);
     this._sanitizedValue = get(this, 'value') || get(this, 'checked');
+
+    for (let key in this.attrs) {
+      if (!NON_ATTRIBUTE_BOUND_PROPS[key]) {
+        this.attributeBindings.push(key);
```
Comment (poteto) [inline addon/components/one-way-input.js]: Don't do this, the array is shared

**Comment by martndemus** [comment]
Fixed by #67 and #68

**Pull request closed without merging**